Client/mods/deathmatch/logic/CClientColPolygon.cpp

```diff
@@ -69,28 +69,20 @@ void CClientColPolygon::SetPosition(const CVector& vecPosition)
     // Add queued collider refresh for v1.1
 }
 
-bool CClientColPolygon::AddPoint(CVector2D vecPoint, int iPointIndex)
+bool CClientColPolygon::AddPoint(CVector2D vecPoint)
 {
-    if (iPointIndex < 0)
-    {
-        m_Points.push_back(vecPoint);
-    }
-    else
-    {
-        if (iPointIndex >= m_Points.size())
-            return false;
-
-        m_Points.insert(m_Points.begin() + iPointIndex, vecPoint);
-    }
+    m_Points.push_back(vecPoint);
+    CalculateRadius(vecPoint);
+    return true;
+}
 
-    CVector2D vecDistance = vecPoint - m_vecPosition;
-    float     fDistance = vecDistance.Length();
+bool CClientColPolygon::AddPoint(CVector2D vecPoint, unsigned int uiPointIndex)
+{
+    if (uiPointIndex > m_Points.size())
+        return false;
```

Comment:
Does this now need to return false if `uiPointIndex < 0`?

Reply:
The type of `uiPointIndex` is unsigned int. Let's say if the value of `uiPointIndex` is -1 when calling AddPoint method, in hex, it will be 0xFFFFFFFF, so the logic is correct. 🙂